### PR TITLE
Bluetooth: host: Handle PA sync cancel event

### PR DESCRIPTION
**Pull request by Thalley**
Fixes a issue when cancelling a PA sync and a shell callback issue.

### CODE REVIEW

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -7481,15 +7486,20 @@ int bt_le_per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync)
...
+		}
 	} else if (get_pending_per_adv_sync() == per_adv_sync) {
 		err = bt_le_per_adv_sync_create_cancel(per_adv_sync);
+		/* Delete of the per_adv_sync will be done in the event
+		 * handler when cancelling */
 	}
 
 	if (err) {
 		return err;
 	}
```
Comment (cvinayak) [inline subsys/bluetooth/host/hci_core.c:7502]:
```suggestion
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/bt.c [hunk 2]
```diff
@@ -1518,7 +1527,10 @@ static int cmd_per_adv_sync_create(const struct shell *shell, size_t argc,
 
 	create_params.options = options;
 
-	bt_le_per_adv_sync_cb_register(&per_adv_sync_cb);
+	if (!cb_registered) {
+		bt_le_per_adv_sync_cb_register(&per_adv_sync_cb);
+	}
+	cb_registered = true;
```
Comment (cvinayak) [inline subsys/bluetooth/shell/bt.c:1533]: You did not delete this after adding in bt_ready.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/bt.c (hunk 2 above)
Comment (Thalley) [reply subsys/bluetooth/shell/bt.c]: Good catch :)

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 3]
```diff
@@ -4674,6 +4674,16 @@ static void le_per_adv_sync_established(struct net_buf *buf)
...
 
+	if (evt->status == BT_HCI_ERR_OP_CANCELLED_BY_HOST) {
+		/* Cancelled locally, don't call CB */
+		if (pending_per_adv_sync) {
+			per_adv_sync_delete(pending_per_adv_sync);
+		} else {
+			BT_ERR("Unexpected per adv sync cancelled event");
+		}
+		return;
```
Comment (jhedberg) [inline subsys/bluetooth/host/hci_core.c:4684]: Minor, but to be consistent with the subsystem-internal style: add an empty line after the `{`

**Review by joerchan — APPROVED** [review]

**Pull request merged**